homeassistant/components/home_connect/sensor.py

```diff
@@ -1,29 +1,189 @@
 """Provides a sensor for Home Connect."""
 
-from datetime import datetime, timedelta
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import date, datetime, timedelta
+from decimal import Decimal
 import logging
 from typing import cast
 
-from homeassistant.components.sensor import SensorDeviceClass, SensorEntity
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import CONF_ENTITIES
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
 import homeassistant.util.dt as dt_util
 
+from .api import ConfigEntryAuth, HomeConnectDevice
 from .const import (
+    ATTR_DEVICE,
     ATTR_VALUE,
+    BSH_EVENT_PRESENT_STATE_OFF,
     BSH_OPERATION_STATE,
     BSH_OPERATION_STATE_FINISHED,
     BSH_OPERATION_STATE_PAUSE,
     BSH_OPERATION_STATE_RUN,
+    COFFEE_EVENT_BEAN_CONTAINER_EMPTY,
+    COFFEE_EVENT_DRIP_TRAY_FULL,
+    COFFEE_EVENT_WATER_TANK_EMPTY,
     DOMAIN,
+    REFRIGERATION_EVENT_DOOR_ALARM_FREEZER,
+    REFRIGERATION_EVENT_DOOR_ALARM_REFRIGERATOR,
+    REFRIGERATION_EVENT_TEMP_ALARM_FREEZER,
+    REFRIGERATION_STATUS_DOOR_CHILLER,
+    REFRIGERATION_STATUS_DOOR_FREEZER,
+    REFRIGERATION_STATUS_DOOR_REFRIGERATOR,
 )
 from .entity import HomeConnectEntity
 
 _LOGGER = logging.getLogger(__name__)
 
 
+def format_state_attr(state: str) -> str:
+    """Format state values to attribute values."""
+    return state.rsplit(".", maxsplit=1)[-1]
+
+
+@dataclass(frozen=True, kw_only=True)
+class HomeConnectSensorEntityDescription(SensorEntityDescription):
+    """Entity Description class for binary sensors."""
+
+    device_class: SensorDeviceClass | None = SensorDeviceClass.ENUM
+    value_fn: Callable[[dict], StateType | date | datetime | Decimal]
+    exists_fn: Callable[[HomeConnectDevice], bool]
+
+
+SENSORS: tuple[HomeConnectSensorEntityDescription, ...] = (
+    HomeConnectSensorEntityDescription(
+        key="Chiller Door",
+        options=["Closed", "Open"],
```

Comment:
If we only have 2 options, shouldn't this be a binary sensor?

Reply:
I had as an `ENUM` state since the API returns an `ENUM` type, though since only two values are ever returned and there's a `DOOR` `device_class`, making it a binary_sensor makes more sense.